quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// s2n-quic's default implementation of the datagram component
+
+use crate::{
+    datagram::unreliable_datagram::{ConnectionInfo, Endpoint, Packet, Receiver, Sender},
+    time::Timestamp,
+};
+use alloc::collections::VecDeque;
+use bytes::Bytes;
+use core::time::Duration;
+
+#[derive(Debug, Default)]
+pub struct Disabled;
+
+// Default capacity is zero right now since the feature is currently disabled.
+const DEFAULT_CAPACITY: usize = 0;
+
+impl Endpoint for Disabled {
+    type Sender = DefaultSender;
+    type Receiver = DisabledReceiver;
+
+    fn create_connection(&mut self, _info: &ConnectionInfo) -> (Self::Sender, Self::Receiver) {
+        let queue = VecDeque::with_capacity(DEFAULT_CAPACITY);
+        (
+            DefaultSender {
+                queue,
+                prioritize_datagrams: true,
+            },
+            DisabledReceiver,
+        )
+    }
+}
+
+pub struct DefaultSender {
+    queue: VecDeque<Datagram>,
+    prioritize_datagrams: bool,
+}
+
+struct Datagram {
+    data: Bytes,
+    creation_time: Timestamp,
```

Comment:
The problem with including this on the Datagram struct is the application won't have access to the clock at the time of pushing. We may need to, instead, compute the creation time as soon as we get a transmission callback. It will be _a bit_ later but it should be sub milliseconds.

Reply:
I think the Datagram struct would still contain a creation_time variable in this case, no? It's creation time per datagram, even if the clock starts later.

Reply:
It would. It's just a matter of it being computed lazily vs computed by the application. So the type may need to be `creation_time: Option<Timestamp>` instead.